src/Native/System.Native/pal_runtimeinformation.cpp

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include "pal_runtimeinformation.h"
+#include "pal_types.h"
+#include <stdio.h>
+#include <sys/utsname.h>
+
+extern "C" const char* GetUnixVersion()
+{
+    struct utsname _utsname;
+    size_t length = _UTSNAME_LENGTH * 3 + 2;
+    char * version = new char[length];
+    if (uname(&_utsname) != -1)
+    {
+        snprintf(version, length, "%s %s %s", _utsname.sysname, _utsname.release, _utsname.version);
+    }
+
+    return version;
+}
+
+/* Returns an int representing the OS Architecture:
+ 0 - x86
+ 1 - x64
+ 2 - ARM */
+extern "C" int32_t GetUnixArchitecture()
+{
+#if defined(ARM)
+    return 2;
+#elif defined(BIT64)
+    return 1;
```

Comment:
Removed the dependency on ```BIT64``` and ```BIT32``` defines, added a custom define (```X64```) for this API purposes, will add ```X86``` and ```ARM64```, if we start building for those.